### PR TITLE
Command localizations don't use camelcase when creating a command with slashcommand builder

### PR DESCRIPTION
**Pull request by schblondie**
**Please describe the changes this PR makes and why it should be merged:**
See #9083 

**Status and versioning classification:**
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed) - Not sure if breaking change, because technically object keys changed

### CODE REVIEW

File: packages/discord.js/src/structures/ApplicationCommand.js [hunk 1]
```diff
@@ -70,12 +70,12 @@ class ApplicationCommand extends Base {
...
     }
 
-    if ('name_localizations' in data) {
+    if ('nameLocalizations' in data) {
       /**
        * The name localizations for this command
        * @type {?Object<Locale, string>}
        */
-      this.nameLocalizations = data.name_localizations;
+      this.nameLocalizations = data.nameLocalizations;
```
Comment (almeidx) [inline packages/discord.js/src/structures/ApplicationCommand.js:78]: The Discord API data comes with camel-cased keys

**Comment by Jiralite** [comment]
Builders is not solely used with discord.js, which makes this pull request incorrect. In addition, changing the casing in discord.js is also wrong, as we are receiving that data from Discord.

File: packages/discord.js/src/structures/ApplicationCommand.js (hunk 1 above)
Comment (schblondie) [reply packages/discord.js/src/structures/ApplicationCommand.js:78]: This is true, I apologize.
But when you compare line 78 with line 88 you can see that the `this.nameLocalized` is also in correct camel case even when getting the non camel case value `data.name_localized`

**Comment by schblondie** [comment]
### I adjusted the code making sure it doesn't conflict with the Discord API. 
Now the SlashCommandBuilder Data is in `camelCase` which doesn't matter when accessing the API as the functions `transformCommand()` (packages\discord.js\src\managers\ApplicationCommandManager.js - line 233) and `transformOption()` (packages\discord.js\src\structures\ApplicationCommand.js - line 549) make sure that the keys are in the appropriate case.

**Pull request closed without merging**